seisidd/experiment.py

```diff
@@ -401,16 +544,845 @@ def scan_closure():
+            beam.l3.l1y     = sim.SynAxis(name='l3y')
+            beam.l3.l1z     = sim.SynAxis(name='l3z')
+            beam.l3.l1rot   = sim.SynAxis(name='l3rot')
+            beam.l3.l1tx    = sim.SynAxis(name='l3tx')
+            beam.l3.l1tz    = sim.SynAxis(name='l3tz')
+            #   Focus lens 4
+            beam.l4         = MotorBundle(name='l4')
+            beam.l4.l1x     = sim.SynAxis(name='l4x')
+            beam.l4.l1y     = sim.SynAxis(name='l4y')
+            beam.l4.l1z     = sim.SynAxis(name='l4z')
+            beam.l4.l1rot   = sim.SynAxis(name='l4rot')
+            beam.l4.l1tx    = sim.SynAxis(name='l4tx')
+            beam.l4.l1tz    = sim.SynAxis(name='l4tz')
+
+            return beam
+
+    @staticmethod
+    def get_nfstage(mode):
+        """return nfstage based on given mode"""
+        if mode.lower() in ['dryrun', 'production']:
+            nfstage = StageAero(name='nfstage')
+        elif mode.lower() == 'debug':
+            # NOTE:
+            #    Moving a synthetic motor will lead to some really strange
+            #    issue.  This could be related to the API change in the
+            #    synAxis.
+            from ophyd import sim
+            from ophyd import MotorBundle
+            nfstage         = MotorBundle(name="nfstage")
+            nfstage.kx      = sim.SynAxis(name='kx')
+            nfstage.ky      = sim.SynAxis(name='ky')
+            nfstage.kz      = sim.SynAxis(name='kz')
+            nfstage.kx_tilt = sim.SynAxis(name='kx_tilt')
+            nfstage.kz_tilt = sim.SynAxis(name='kz_tilt')
+
+            nfstage.rot    = sim.SynAxis(name='rot')
+
+            nfstage.x_base  = sim.SynAxis(name='x_base')
+            nfstage.y_base  = sim.SynAxis(name='y_base')
+            nfstage.z_base  = sim.SynAxis(name='z_base')
+            
+        else:
+            raise ValueError(f"Invalide mode -> {mode}")
+        return nfstage
+
+    @staticmethod
+    def get_flycontrol(mode):
+        if mode.lower() == 'debug':
+            # TODO: need better simulated motors
+            from ophyd import sim
+            psofly = sim.flyer1
+        elif mode.lower() in ['dryrun', 'production']:
+            psofly = EnsemblePSOFlyDevice("PV_FLY", name="psofly")
+        else:
+            raise ValueError(f"Invalide mode, {mode}")
+        return psofly
+
+    @staticmethod
+    def get_detector(mode):
+        if mode.lower() == 'debug':
+            # TODO: need better simulated detectors
+            from ophyd import sim
+            det = sim.noisy_det
+        elif mode.lower() in ['dryrun', 'production']:
+            det = PointGreyDetector6IDD("PV_DET", name='det')
+            # check the following page for important information
+            # https://github.com/BCDA-APS/use_bluesky/blob/master/notebooks/sandbox/images_darks_flats.ipynb
+            #
+            epics.caput("PV_DET:cam1:FrameType.ZRST", "/exchange/data_white_pre")
```

Comment:
for different experiment, we might need to set different layout (folder names) through the same PVs.  This is just something that we have to keep in mind.

Reply:
Sure, I'm much less familiar with this.  Let me know what you need to change if that happens.